### PR TITLE
backoffice: add location/intenal location form

### LINKED ISSUE #492: forms: create, update locations
Implement a form so the user can create and update `locations` and `internal_locations`, take the access rights front and backend into consideration.

addresses #464

### PR DESCRIPTION
**Pull request by zzacharo**
closes #492

### CODE REVIEW

File: invenio_app_ils/records/loaders/schemas/json/location.py [hunk 1]
```diff
@@ -7,16 +7,23 @@
...
 
+    class Meta:
+        """Meta attributes for the schema."""
+
+        from marshmallow import EXCLUDE
+        unknown = EXCLUDE
+
     pid = PersistentIdentifier()
     name = fields.Str(required=True)
+    address = fields.Str(required=True)
```
Comment (ntarocco) [inline invenio_app_ils/records/loaders/schemas/json/location.py]: remove `required`? Not sure we want it mandatory...

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationList/components/InternalLocationList/InternalLocationList.js [hunk 2]
```diff
@@ -20,7 +26,7 @@ export default class InternalLocationList extends Component {
   }
 
   handleOnRefClick(itemPid) {
-    // TODO: EDITOR, implement edit form
+    // TODO: navigate to item edit form
```
Comment (ntarocco) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationList/components/InternalLocationList/InternalLocationList.js:29]: is this missing on purpose?

**Review by ntarocco — DISMISSED** [review]

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationList/components/InternalLocationList/InternalLocationList.js (hunk 2 above)
Comment (zzacharo) [reply invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationList/components/InternalLocationList/InternalLocationList.js:29]: yes I don't have the item edit form yet :)

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationForms/InternalLocationEditor/InternalLocationEditor.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    this.state = {
+      data: {},
+      isLoading: true,
+      error: {},
+    };
+  }
+
+  fetchInternalLocation = async ilocationPid => {
+    try {
+      const response = await internalLocationApi.get(ilocationPid);
```
Comment (kpsherva) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationForms/InternalLocationEditor/InternalLocationEditor.js:20]: shouldn't this be in the actions?

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationForms/InternalLocationEditor/components/InternalLocationForm/InternalLocationForm.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    this.successSubmitMessage = props.successSubmitMessage;
+    this.title = props.title;
+  }
+  prepareData = data => {
+    return _.pick(data, ['name', 'location_pid', 'physical_location', 'notes']);
+  };
+
+  onSubmit = async (values, actions) => {
+    try {
+      actions.setSubmitting(true);
```
Comment (kpsherva) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/Location/LocationForms/InternalLocationEditor/components/InternalLocationForm/InternalLocationForm.js:27]: this also usually is in the `actions.js` do we change the concept?

**Review by kpsherva — DISMISSED** [review]

**Pull request merged**